File: src/Enterprise/C#/Example1.cs
using System;

// Abstract Product
public interface Button {
    void Render();
}

public interface Checkbox {
    void Render();
}

// Concrete Product
public class DarkButton : Button {
    public void Render() {
        Console.WriteLine("Dark Button");
    }
}

public class LightButton : Button {
    public void Render() {
        Console.WriteLine("Light Button");
    }
}

public class DarkCheckbox : Checkbox {
    public void Render() {
        Console.WriteLine("Dark Checkbox");
    }
}

public class LightCheckbox : Checkbox {
    public void Render() {
        Console.WriteLine("Light Checkbox");
    }
}

// Abstract Factory
public interface UIFactory {
    Button CreateButton();
    Checkbox CreateCheckbox();
}

// Concrete Factory
public class DarkFactory : UIFactory {
    public Button CreateButton() {
        return new DarkButton();
    }

    public Checkbox CreateCheckbox() {
        return new DarkCheckbox();
    }
}

public class LightFactory : UIFactory {
    public Button CreateButton() {
        return new LightButton();
    }

    public Checkbox CreateCheckbox() {
        return new LightCheckbox();
    }
}

public class Example1 {
    public static void CreateUIComponents(UIFactory factory) {
        Button button = factory.CreateButton();
        Checkbox checkbox = factory.CreateCheckbox();
        button.Render();
        checkbox.Render();
    }

    public static void Main(string[] args) {
        CreateUIComponents(new DarkFactory());
        CreateUIComponents(new LightFactory());
    }
}

File: src/Enterprise/Java/Example1.java
// Abstract Product
interface Button {
    void render();
}

interface Checkbox {
    void render();
}

// Concrete Product
class DarkButton implements Button {
    @Override
    public void render() {
        System.out.println("Dark Button");
    }
}

class LightButton implements Button {
    @Override
    public void render() {
        System.out.println("Light Button");
    }
}

class DarkCheckbox implements Checkbox {
    @Override
    public void render() {
        System.out.println("Dark Checkbox");
    }
}

class LightCheckbox implements Checkbox {
    @Override
    public void render() {
        System.out.println("Light Checkbox");
    }
}

// Abstract Factory
interface UIFactory {
    Button createButton();
    Checkbox createCheckbox();
}

// Concrete Factory
class DarkFactory implements UIFactory {
    @Override
    public Button createButton() {
        return new DarkButton();
    }
    
    @Override
    public Checkbox createCheckbox() {
        return new DarkCheckbox();
    }
}

class LightFactory implements UIFactory {
    @Override
    public Button createButton() {
        return new LightButton();
    }
    
    @Override
    public Checkbox createCheckbox() {
        return new LightCheckbox();
    }
}

public class Example1 {
    public static void createUIComponents(UIFactory factory) {
        Button button = factory.createButton();
        Checkbox checkbox = factory.createCheckbox();
        button.render();
        checkbox.render();
    }

    public static void main(String[] args) {
        createUIComponents(new DarkFactory());
        createUIComponents(new LightFactory());
    }
}
